### PR TITLE
bpf: compile out service lookup entirely on kubeProxyReplacement=disa…

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg. /cc @brb

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10060)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: daemon/daemon_main.go [hunk 1]
```diff
@@ -1426,6 +1426,10 @@ func initKubeProxyReplacementOptions() {
 		log.Fatalf("Invalid value for --%s: %s", option.KubeProxyReplacement, option.Config.KubeProxyReplacement)
 	}
 
+	if option.Config.DisableK8sServices {
+		option.Config.KubeProxyReplacement = option.KubeProxyReplacementDisabled
```
Comment (brb) [inline daemon/daemon_main.go]: Nit: let's log a warning here if previously kube-proxy-replacement != disabled.

**Review by brb — APPROVED** [review]

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
provision error

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
provision error, retrying

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**